src/MobileRedirection.php

```diff
@@ -64,7 +64,15 @@ public function register() {
 		add_filter( 'amp_default_options', [ $this, 'filter_default_options' ] );
 		add_filter( 'amp_options_updating', [ $this, 'sanitize_options' ], 10, 2 );
 
-		if ( AMP_Options_Manager::get_option( Option::MOBILE_REDIRECT ) && ! amp_is_canonical() ) {
+		$is_mobile_request = AMP_Options_Manager::get_option( Option::MOBILE_REDIRECT );
+		$sandboxing_level  = amp_get_sandboxing_level();
+
+		// Add alternative link if mobile redirection is enabled or sandboxing level is set to loose or moderate.
+		if ( $is_mobile_request || ( 1 === $sandboxing_level || 2 === $sandboxing_level ) ) {
```

Comment:
One more condition is needed: we don't want to have the link added when on Standard mode (if `amp_is_canonical()`) as otherwise this alternate link will just be a self-link.

```suggestion
		if ( ! amp_is_canonical() && ( $is_mobile_request || ( 1 === $sandboxing_level || 2 === $sandboxing_level ) ) ) {
```

Also, it needs to be omitted when on an AMP page. Right now I'm seeing the link even on an AMP page. We can't check `amp_is_request()` here since it's too early, so this could be addressed by adding the following to the top of the `add_mobile_alternative_link` method:

```php
if ( amp_is_request() ) {
	return;		
}
```

Reply:
> We can't check amp_is_request() here since it's too early, so this could be addressed by adding the following to the top of the add_mobile_alternative_link method:

While adding in top of `add_mobile_alternative_link` it's calling `amp_is_available` early and hence resulting in an error. I have added a different workaround in https://github.com/ampproject/amp-wp/pull/7288/commits/2d23020c9e63cfe48b50f030486e7a98fcc2b37f to make sure it's get called at the right time.